docs/modules/ROOT/pages/reference/arch-control-plane.adoc

```diff
@@ -0,0 +1,162 @@
+
+image::universal-control-plane.drawio.svg[]
+
+We inherit the https://kb.vshn.ch/appuio-cloud/references/architecture/control-api.html[APPUiO Control API^], extend it for VAMP and rebrand it into the VSHN Portal.
+
+== Terminology
+
+Central Control Plane (CCP)::
+This is the aggregated view and entry point for the user.
+It provides the mechanisms for the multi-tenant, multi-csp self-service portal.
+
+CSP Control Plane (CSPCP)::
+This runs the Crossplane and VAMP Control Plane which manages all the service instances on worker clusters.
+
+Worker Cluster::
+Clusters running the actual workload, managed by the CSPCP.
+Services are directly exposed to the end-user.
+
+Service::
+A managed service which can be instantiated (ordered) and parametrized according to the specification.
+
+Plan::
+A pre-defined set of parameters for a service.
+
+Crossplane specific terms::
+* Compositions - A template to define how to create resources.
+* Composite Resource Definition (`XRD`) - A custom API specification.
+* Composite Resource (`XR`) - Created by using the custom API defined in a Composite Resource Definition. XRs use the Composition template to create new managed resources.
+* Claims (`XRC`) - Like a Composite Resource, but with namespace scoping.
+* Managed Resource (`MR`) - Represents an external service in a Provider.
+* Providers enable Crossplane to provision infrastructure on an external service. Providers create new Kubernetes APIs and map them to external APIs.
+
+== Organization Origin
+
+This enhances https://kb.vshn.ch/appuio-cloud/references/architecture/control-api-org.html[APPUiO Control API: Organization^] with the aspect of the origin of an organization.
+The origin specifies from where an organization originates, either from VSHN or a CSP like Exoscale or cloudscale.
+This origin is then used in further processes, like filtering of resources or access control.
+
+* Organizations are marked from where they are coming from
+* E.g. we put a label or annotation on the Namespace and expose this in the virtual resource under `.spec.origin`
+
+=== Origin Metadata
+
+An origin has a configuration which is used to parametrize further processes.
+This is done in a CRD `OriginConfig`.
+
+An `OriginConfig` contains:
+
+* A reference to the `ProviderConfig`
+* A list of available services (XRDs)
+* A list of available zones
+
+[source,yaml]
+----
+apiVersion: appcat.vshn.io/v1
+kind: OriginConfig
+metadata:
+  name: exoscale
+spec:
+  providerConfigRef:
+    name: exoscale-cspcp
+  zones:
+    - de-fra-1
+    - ch-gva-2
+  XRDRefs:
+    - VSHNPostgreSQL
+    - VSHNMariaDB
+----
+
+This configuration can be overwritten per organization namespace, for example to give access to "beta" services or additional zones.
+
+== Service Catalog
+
+A service is exposed by a https://docs.crossplane.io/v1.18/concepts/composite-resource-definitions/[Crossplane Composite Resource Definitions (XRD)^].
```

Comment:
We should think about having compositions as the single point of truth.

An XRD can implement multiple compositions which in turn could implement different specifics. Like another zone or even provider. That's why `kubectl get appcat` currently queries compositions and not the XRDs.

You even mention below that the true differentiation lies in the composition.

This way we can have a single generic XRD to describe any arbitrary service, including the spec for a claim.
Then the composition will just push that embedded claim to the right control plane.

Reply:
From the discussion here more details how I think this could look (conrete names TBD):

```

apiVersion: vamp.vshn.io/v1
kind: CenrtralServiceClaim # CSC to keep the acronyms :D
metadata:
  name: serviceA #probably generated from something
  namespace: orgNamespace
spec:
  claimContent:
    apiVersion: vshn.appcat.vshn.io/v1
    kind: VSHNPostgreSQL
    metadata:
       name: serviceA
       namespace: orgNamespace
     *RESTOFCLAIM* 
  compositionRef:
    name: providerA-zoneB-serviceName.vamp.vshn.io
  writeConnectionSecretToRef:
    name: serviceA
    namespace: orgNamespace
```

The composition `providerA-zoneB-serviceName.vamp.vshn.io` will then:
* Create a kube object with the right `providerConfig` for that CSPCP
* Maybe even ensure that the org namespace exists in that CSPCP 
* Provide the connection details for that instance back on the CCP

Reply:
Thinking even further for the future:

If we're at a point where we have even multiple CSPCPs for a signle zone in for a single CSP, we could then make the `providerA-zoneB-serviceName.vamp.vshn.io` smarter and do actual loadbalancing. E.g. it would connect to the CSPCPs via kubeclient and decide according to some criteria which CSPCP to use.

Reply:
> From the discussion here more details how I think this could look (conrete names TBD):
> 
> ```
> 
> apiVersion: vamp.vshn.io/v1
> kind: CenrtralServiceClaim # CSC to keep the acronyms :D
> metadata:
>   name: serviceA #probably generated from something
>   namespace: orgNamespace
> spec:
>   claimContent:
>     apiVersion: vshn.appcat.vshn.io/v1
>     kind: VSHNPostgreSQL
>     metadata:
>        name: serviceA
>        namespace: orgNamespace
>      *RESTOFCLAIM* 
>   compositionRef:
>     name: providerA-zoneB-serviceName.vamp.vshn.io
>   writeConnectionSecretToRef:
>     name: serviceA
>     namespace: orgNamespace
> ```
> 
> The composition `providerA-zoneB-serviceName.vamp.vshn.io` will then:
> 
>     * Create a kube object with the right `providerConfig` for that CSPCP
> 
>     * Maybe even ensure that the org namespace exists in that CSPCP
> 
>     * Provide the connection details for that instance back on the CCP

Doesn't that break the API compatibility? As an end-user, I want to put the same `VSHNPostgreSQL` in the CCP like I do in the CSPCP. By choosing the `compositionRef` I tell the CCP where I would like to make this service happen.